apps/site/assets/ts/schedule/components/schedule-finder/daily-schedule/TableRow.tsx

```diff
@@ -1,50 +1,39 @@
-import React, { useEffect, useState, useReducer, ReactElement } from "react";
-import { reducer } from "../../../helpers/fetch";
-import { Journey, EnhancedJourney, TripInfo } from "../__trips";
-import { modeIcon } from "../../../helpers/icon";
-import { breakTextAtSlash } from "../../../helpers/text";
-import { UserInput } from "../../components/__schedule";
+import React, { useEffect, useState, ReactElement } from "react";
+import { modeIcon } from "../../../../helpers/icon";
+import { breakTextAtSlash } from "../../../../helpers/text";
+import useFetch, { isNotStarted } from "../../../../helpers/use-fetch";
+import { isACommuterRailRoute } from "../../../../models/route";
+import { UserInput } from "../../__schedule";
+import { Journey, TripInfo } from "../../__trips";
 import AccordionRow from "./AccordionRow";
+import TripDetails from "./TripDetails";
 
 interface TableRowProps {
   input: UserInput;
-  journey: Journey | EnhancedJourney;
+  journey: Journey;
   isSchoolTrip: boolean;
   anySchoolTrips: boolean;
 }
 
 interface AccordionProps {
   input: UserInput;
-  journey: Journey | EnhancedJourney;
+  journey: Journey;
   contentComponent: () => ReactElement<HTMLElement>;
 }
 
-type fetchAction =
-  | { type: "FETCH_COMPLETE"; payload: TripInfo }
-  | { type: "FETCH_ERROR" }
-  | { type: "FETCH_STARTED" };
-
-export const fetchData = (
+// Exported solely for testing
+export const fetchTripInfo = (
```

Comment:
I'm not totally sure if it's doable but could you use this newly created `fetchTripInfo` in `UpcomingDepartures.tsx`?

Reply:
Yeah, I do want to get there. The tricky thing about doing it right now is that the data from upcoming departures is requested individually mapping over a list, not just from one request. So we'll either have to make these helpers work with that, or maybe better yet make a route on the server that lets the client request data for that whole list in a single request. But it was enough that I didn't want to take it on in this PR.

Reply:
Sure, thanks for clarifying :smiley: